examples/source_examples/accelo_api_connector_multithreading_enabled/connector.py

```diff
@@ -0,0 +1,763 @@
+        int_fields = [
+            "id", "weighting", "value_weighted", "won_by_id",
+            "cancelled_by_id", "abandoned_by_id", "contact",
+            "manager", "prospect_type", "status", "prospect_probability",
+            "affiliation"
+        ]
+        for field in int_fields:
+            value = record.get(field)
+            if value is not None:
+                try:
+                    record[field] = int(value)
+                except (ValueError, TypeError):
+                    log.warning(f"Could not convert field {field} with value '{value}' to int")
+                    record[field] = None
+
+        # Convert float fields
+        float_fields = ["value", "progress"]
+        for field in float_fields:
+            value = record.get(field)
+            if value is not None:
+                try:
+                    record[field] = float(value)
+                except (ValueError, TypeError):
+                    log.warning(f"Could not convert field {field} with value '{value}' to float")
+                    record[field] = None
+
+        # Convert date fields
+        date_fields = [
+            "date_created", "date_actioned", "date_due",
+            "date_last_interacted", "date_modified"
+        ]
+        convertDateFields(date_fields, record)
+
+        # Convert 'success' field to boolean
+        value = record.get('success')
+        if value is not None:
+            try:
+                if isinstance(value, bool):
+                    record['success'] = value
+                elif isinstance(value, str):
+                    if value.lower() == 'yes' or value.lower() == 'true':
+                        record['success'] = True
+                    elif value.lower() == 'no' or value.lower() == 'false':
+                        record['success'] = False
+                    else:
+                        raise ValueError(f"Unexpected string value for success: {value}")
+                else:
+                    raise ValueError(f"Unexpected type for success: {type(value)}")
+            except (ValueError, TypeError) as e:
+                log.warning(f"Could not convert field 'success' with value '{value}' to bool: {e}")
+                record['success'] = None
+
+        return record
+
+    yield from sync_entity(
+        entity_name="prospects",
+        access_token=access_token,
+        fields=prospect_fields,
+        last_sync_key="last_prospect_sync",
+        process_record=process_prospect_record,
+        date_field="date_modified",
+        timeout=SYNC_TIMEOUT,
+        batch_size=BATCH_SIZE
+    )
+
+
+def sync_jobs(access_token):
+    """
+    Sync job data from Accelo API.
+    """
+    job_fields = [
+        "id", "title", "status", "standing", "date_created", "date_started",
+        "date_due", "date_completed", "date_modified", "manager", "company",
+        "contact", "description", "value"
+    ]
+
+    def process_job_record(record):
+        # Convert integer fields
```

Comment:
You can create a common method for converting integer and float fields as well.

Reply:
Done.